flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java

```diff
@@ -88,7 +88,8 @@ public void init() throws Exception {
 					getStreamStatusMaintainer(),
 					this.headOperator,
 					getEnvironment().getMetricGroup().getIOMetricGroup(),
-					inputWatermarkGauge);
+					inputWatermarkGauge,
+					getEnvironment().getTaskInfo().getTaskNameWithSubtaskAndID());
```

Comment:
Maybe we could use `getEnvironment().getExecutionId()` and `getEnvironment().getTaskInfo().getTaskName()` to construct the `getTaskNameWithSubtaskAndID()`. Then we don't have to touch `TaskInfo` at all.

Reply:
Yes, that could also work. The only concern is we need do this multiple times in different components which need task name. Actually I think the task name with different info should belong to the scope of `TaskInfo` if we are caring about the previous implementation limit. WDYT?

Reply:
I think `TaskInfo` has the purpose to encapsulate some user facing information about the task. That's why it is located in `flink-core`. The `ExecutionAttempID` and also the `AllocationID` are runtime specific concepts which should not be part of this class imo. Thus, I would strongly suggest to not mix concerns and cleanly separate the different aspects. Which components actually need `getTaskNameWithSubtaskAndID()`? As far as I can tell, it is just the `UnionInputGate` as well as the `BarrierBuffer`.

Reply:
Agree this suggestion, and I could adjust it.